src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
+    return next.handle().pipe(
+      map((data) => {
+        const response = {
+          isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
```

Comment:
PullProofAI

Consider simplifying the condition for `isSuccess` for better readability.
```diff
-           isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
+           isSuccess: statusCode >= 200 && statusCode < 300,
```


<!-- This is an auto-generated comment by PullProofAI -->